Database-Features/removing-corrupt-data.md

```diff
@@ -0,0 +1,91 @@
+# Identifying and Removing Corrupt Data
+
+## Scope
+
+In very rare occasions, a chain of actions may cause some data in the database to become corrupt. This article gives some tools you can use to help identify if there is corrupt data or not. 
+
+**Corrupt data occurs very rarely, even if your issues match the symptoms in this article. You should only follow the below instructions after discussion with Exasol support.**
+
+## Diagnosis
+
+If something in the database is corrupt, it can manifest itself in several different ways. For example:
+* Some queries may cause the database to suddenly restart. The queries may reference certain objects in the database which are corrupt.
+* Backups are unable to be written.
+
+## Explanation
+
+Data is stored column-wise in Exasol, which means that if certain data blocks are removed or not accessible, then any attempt to access that block will crash the database. This includes queries reading certain columns, indexes, statistics, or other data structure, as well as backups. In this case, the database is in an inconsistent state and requires intervention.
+
+## Recommendation
+
+Depending on the setup of the database and the effort to perform the below tasks, there are the following options:
+
+### 1. Restore from Remote Backup
+If we identify that the database is an inconsistent state and there is a recent backup stored remotely (from before the event which caused the corruption), than a quick option is to restore the database from the latest available backup. This action would cause a downtime which corresponds to the time required to write the full and incremental backups. In addition, any data which was added to the database since the latest remote backup would need to be reloaded into the database. 
+
+Note: Local backups which were written before the event may not be reliable. There is no way to verify if the data blocks from the local backup were corrupted or not. For this reason, the backup to restore from should be stored remotely. 
```

Comment:
Is there an issue to improve this ? 

It reads "horrible" that local backups are just written with corrupted blocks but remote backups are not.

I doubt a bit the correctness of the statement as well. From EXA_STATISTICS.EXA_SYSTEM_EVENTS we see BACKUP_ABORTED for failed (local) backups in such cases. This symptom showed up in the corruption support cases - somewhat independent from local or remote backup differentiation. 
But maybe you are referring to EXAoperation ?

Reply:
I'm referring to the following situation. 
1. Customer performs a local L0 backup
2. At some point, (due to hardware issues or whatever), data blocks are removed. 
3. The user is not able to verify if data blocks were removed from the backup as well unless they try to restore the database (which I'm not sure would work or not in this case). 

I'm trying to make the point that if there is data corruption, even blocks from a local archive volume might be corrupted without the user knowing it. So the backup may not be reliable. However, a remote backup would always be safe. Does that make sense?

Reply:
I am not sure that I understand "how blocks can get removed from backups". 

No, it does not make sense to me. Remote backups should be no different than local ones.

Reply:
Made changes based on conversation in slack